Sources/AudioKitUI/Visualizations/FFTView.swift

```diff
@@ -77,51 +63,75 @@ public struct FFTView: View {
     private var paddingFraction: CGFloat
     private var includeCaps: Bool
     private var node: Node
-    private var numberOfBars: Int
-    private var minAmplitude: Double
-    private var maxAmplitude: Double
+    private var barCount: Int
+    private var fftValidBinCount: FFTValidBinCount?
+    private var minAmplitude: Float
+    private var maxAmplitude: Float
+    private let defaultBarCount: Int = 64
+    private let maxBarCount: Int = 128
```

Comment:
I pulled down the branch to play around with this. Since this number is so low, it means that any binCount above it will result in higher frequencies not being displayed.

i think what this means is that there is no way around averaging (or some other aggregation) of values to get the full spectrum to be displayed. if we're talking iphones, we only have about 1000 pixels of screen width to work with, so there wouldn't be any way to show the higher bin counts without aggregating.

probably fine for now given that we're already truncating, but i think we should fix this.

Reply:
The binCount addition was my attempt to come up with a simple solution to providing the whole spectrum, if desired.

Please correct me if I'm wrong, but doesn't altering the number of bins returned by the FFTTap to return the same count as the number of bars achieve the result you're looking for? This way you are not throwing away any data and the data returned has already split the bins appropriately without any fancy algorithm on your part.

When the binCount is higher than the barCount,I know we are not displaying the high frequency bins. That is intentional.